tests/model/test_cycle_data.py

```diff
@@ -412,6 +423,79 @@ def test_irrelevant_repos_data_removal(
     ).exists()
 
 
+def test_signature_remover(test_perf_signature, test_perf_signature_2, test_perf_data, monkeypatch):
+    monkeypatch.setattr(taskcluster, 'Notify', MagicMock())
+    cycler = PerfherderCycler(chunk_size=100, sleep_time=0)
+    expired_timestamp = cycler.max_timestamp
+    test_perf_signature_2.last_updated = expired_timestamp
+    test_perf_signature_2.save()
+
+    assert len(PerformanceSignature.objects.all()) == 2
+
+    call_command('cycle_data', 'from:perfherder')
+
+    assert taskcluster.Notify().ping.called_once
+    assert taskcluster.Notify().email.call_count == 2
+    assert len(PerformanceSignature.objects.all()) == 1
+    assert PerformanceSignature.objects.first() == test_perf_signature
+
+
+def test_signature_remover_when_notify_service_is_down(
+    test_perf_signature, test_perf_signature_2, test_perf_data, monkeypatch
+):
+    monkeypatch.setattr(taskcluster, 'Notify', MagicMock())
+    taskcluster.Notify().ping.side_effect = Exception('Email Service is down.')
+
+    call_command('cycle_data', 'from:perfherder')
+
+    assert taskcluster.Notify().ping.called_once
+    assert not taskcluster.Notify().email.called
+    assert len(PerformanceSignature.objects.all()) == 2
+
+
+@pytest.mark.parametrize('signatures_nr', [3, 4, 8, 10])
```

Comment:
'nr' is a Romanian (not English) shorthand for 'number'. We should try to use something different instead. Like 'total' (total_signatgures) or 'amount' etc